Fix for incorrect image sizes in edit mode on image component Banner Variant

Chris1415 opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
The current behavior of image component Banner variant in edit mode is, that the image is rendered full size, which leads to massive overlap with other components on the page. This only happens in Edit mode.
<!-- Apply the label "bug" or "enhancement" as applicable. -->

## Description / Motivation
<!-- Describe your changes in detail -->
1. I added a proper div container arround the image field to let it participate in the following rule set in CSS
```
.image img {
  max-width: 100%;
  height: auto;
}
```
2. I added a conditional rendering of the background image to avoid double rendering the background one AND the one from sc-img tag helper

<!-- Why is this change required? What problem does it solve? -->
This solves the following:
1. The image is not rendered full size in editing anymore
2. There is no double rendering of background image and sc-img anymore in editing

Behavior with applied fix:
![image](https://github.com/user-attachments/assets/e824bbbf-72e8-44cb-be77-261ab18a8581)

<!-- If it fixes an open issue, please link to the issue here. -->
This PR fixes the following reported Bug: Closes https://github.com/Sitecore/xmcloud-starter-dotnet/issues/10


## Terms
<!-- Place an X in the [] to check. -->

<!-- The Code of Conduct helps create a safe space for everyone. We require that everyone agrees to it. -->
- [ x] I agree to follow this project's [Code of Conduct](CODE_OF_CONDUCT.md).

Chris1415 commented:
@robearlam as discussed here is the PR. Also created a bug in advance to link it properly. As this is my first PR for this Project just let me know if anything fundamental is wrong with the approach. Happy to change

robearlam commented:
> @robearlam as discussed here is the PR. Also created a bug in advance to link it properly. As this is my first PR for this Project just let me know if anything fundamental is wrong with the approach. Happy to change

Looks good @Chris1415 I added one small comment.

Also, in the main description if you reference Issue the by stating "Closes #10" then when the PR is closed the Issue will automatically be closed as well

Chris1415 commented on headapps/aspnet-core-starter/Views/Shared/Components/SitecoreComponent/Image.cshtml:
```diff
@@ -1,34 +1,37 @@
             <span class="sc-image-wrapper">
-                @if (Model.IsEditing || string.IsNullOrWhiteSpace(Model.TargetUrl?.Value.Href))
-                {
+            @if (Model?.IsEditing ?? false || string.IsNullOrWhiteSpace(Model.TargetUrl?.Value.Href))
```
Done

Chris1415 commented:
> > @robearlam as discussed here is the PR. Also created a bug in advance to link it properly. As this is my first PR for this Project just let me know if anything fundamental is wrong with the approach. Happy to change
> 
> Looks good @Chris1415 I added one small comment.
> 
> Also, in the main description if you reference Issue the by stating "Closes #10" then when the PR is closed the Issue will automatically be closed as well

Added this one in description as well. Thanks for the heads up

robearlam reviewed (approved):
Looks Good!
